\documentclass[a4paper]{article} 
\usepackage[utf8]{inputenc} 
\usepackage[T1]{fontenc}
\usepackage[english]{babel}
\usepackage[colorlinks = true, 
            urlcolor = black,
            pdfborder = 0 0 0,
            citecolor = red]{hyperref}
\usepackage{rotating}
\SweaveOpts{out.width = "\\maxwidth", size = "footnotesize"} 

\makeatletter
\def\maxwidth{ %
  \ifdim\Gin@nat@width>\linewidth
    \linewidth
  \else
    \Gin@nat@width
  \fi
}
\makeatother

\title{Vignette for package \texttt{resamplediversity}}
\author{Tomaž Skrbinšek}
\date{\today}

\begin{document}

\maketitle

% setup chunk for knitr
<<echo=FALSE>>=
options(width = 60) 
pdf.options(encoding = "CP1250")
#setwd("./resamplediversity/manual_vignette")
load("bine.RData")
source("../R/functions.R")
library(xtable)
@

This vignette documents workflow from our paper^{\cite{skrbinsek2012}}.

\section*{Installing the package}
On Windows, you can install the R package \texttt{diversitycompare} with the following command

<<install_pkg, echo = TRUE, eval = FALSE>>=
install.packages(file.choose(), repos = NULL)
@

whereupon a window will pop-up. You can now select the package binary (.zip). Instead of
\texttt{file.choose()} you can specify a (full) path to the .zip file. Ultimately, you 
can build from the \textbf{source} tarball on any operating system platform.

\vspace{0.4cm}

\noindent Should you have problem installing the package, do not hesitate to contact the author.

<<toolchain, eval = FALSE, tidy = FALSE>>=
install.packages(pkgs = "/diversitycompare.tar.gz",
                 repos = NULL,
                 type = "source")
@

The Dinaric bears are used as the reference population. We can make a quick summary of the data and look at the loci that were used.
<<eval = FALSE>>=
data(dinaric.genotypes)
@
<<look_data, echo = TRUE, eval = TRUE>>=
summary(dinaric.genotypes)
locNames(dinaric.genotypes)
@

To compare genetic diversity indices between two populations, we need to have a common set of loci and provide correction for unequal sample sizes. The latter is especially important for estimates of allelic richness, as this parameter is heavily dependent on sample size (rare alleles will not make it when sample size is low). Expected heterozygosity is much more robust.

We first need a list of common markers. Let's look at a table of diversity parameters from different
 brown bear populations around the world (see table \ref{table:bbs}):
<<eval = FALSE>>=
data(bear.diversity)
bear.diversity
@

<<look_diversity, echo = FALSE, eval = TRUE, results = "asis">>=
print(xtable(bear.diversity, caption = "Table of brown bear diversity data from a number of studies around the world.",
    label = "table:bbs"), table.placement = "!h", caption.placement = "top")
@

Let's compare genetic diversity between Dinaric bears and bears in Kluane, Yukon. They were studied in study 2:
<<eval = FALSE>>=
data(included.studies)
@

<<look_study, echo = TRUE, eval = TRUE>>=
bear.diversity[4, ]
included.studies[included.studies$ID == 2, ]
@

Looking at the original paper by Paetkau et al., the common markers between both populations are G10B, G10C ,G10D, G10L, G10M, G10P, G10X, G1A. We look at the markers in the reference genotypes:
<<look_loci, echo = TRUE, eval = TRUE>>=
locNames(dinaric.genotypes)
@

Genetic diversity study of this population included samples of 28 individuals. We need to subset the locus panel using generic names of loci:
<<look_genericLoci, echo = TRUE, eval = TRUE>>=
loci_na <- c("L02", "L03", "L04", "L06", "L07", "L08", "L09", "L10")
@

We will resample Dinaric genotypes multiple times to the same sample size that was used the Kluane population study (50 samples) using the sampe panel of loci to get comparable genetic diversity indices. This will take a while and produce a lot of relatively useless output from each subsample (omitted here)
<<resample_alaska, echo = TRUE, eval = FALSE, tidy = FALSE>>=
resampled.ar <- subsample.gen(genotypes = dinaric.genotypes,
                              nboots = 1000,
                              nsamps = 50,
                              loci = loci_na)
@

Look at the results:
<<print_alaska, echo = TRUE, eval = TRUE>>=
resampled.ar
@

Now we can calculate diversity ratios between the Dinaric bear population and Kluane bears.
<<div.ratio, echo = TRUE, eval = TRUE, tidy = FALSE>>=
calcDivRat(ref = 6.12, SEref = 0.7, obs = 7.38,
           SEobs = 0.56, type = "A") #allelic richness ratio
calcDivRat(ref = 0.73, SEref = 0.026, obs = 0.76, 
           SEobs = 0.025, type = "He") #heterozygosity ratio
@

We can see that allelic richness is 21\% higher in Kluane, and heterozgosity 4\%.

We can now batch-run the corrections for the entire set of North American populations studied 
by Paetkau et al. using the same locus set. Result can be found in table \ref{table:na.pops}.

<<napops, echo = TRUE, eval = TRUE, tidy = FALSE>>=
na.pops <- bear.diversity[bear.diversity$Study == 1 | bear.diversity$Study == 2, ]
@

<<echo = FALSE, tidy = FALSE, results = "asis">>=
print(xtable(na.pops, caption = "Diversity ratios between Dinaric bear population and Kluane bears.", label = "table:na.pops"),
  caption.placement = "top", table.placement = "!h")
@


The batch run will \textsc{take a long time} and produce a lot of useless output on screen. 
I reduced the number of resamples (\texttt{nboots}) to 100 to keep the computation time reasonable.
In a real study, you would want \texttt{nboots} to be at least 1000.
<<resample_na, echo = TRUE, eval = FALSE, tidy = FALSE>>=
adjusted_na <- runall(N = na.pops$N,
                      genotypes = dinaric.genotypes,
                      loci = loci_na,
                      nboots = 100)
@

\noident Results are presented below.
<<print_na, echo = TRUE, eval = TRUE>>=
#these are resampled value for the reference population, hence prefix "ref".
names(adjusted_na) <- paste("ref", names(adjusted_na), sep = "") 
adjusted_na <- cbind(na.pops, adjusted_na)
adjusted_na
@

% ne gre spravit tabele na eno normalno stran, je preširoka
% bi moral razbit na več sub-data-frameov in vsakega posebej
% z xtable izpisat. kar pa bi bilo verjetno dost nepregledno

%\newpage
%<<echo = FALSE, tidy = FALSE, results = "asis">>=
%print(xtable(adjusted_na, caption = "Results for correction of the entire set of North American populations studies (Paetkau et al.).",
%  label = "table:ad_na"), caption.placement = "top", table.placement = "!h",
%  include.rownames = FALSE, floating.environment = "sidewaystable", scalebox = 0.8)
%@

\noindent We can now calculate diversity ratios:
<<divratios_na, echo = TRUE, eval = TRUE, tidy = FALSE>>=
Ar.na <- with(adjusted_na, 
              calcDivRat(ref = refA, SEref = refSEA, obs = A,
                         SEobs = SEA, type = "A"))
Her.na <-  with(adjusted_na,
                calcDivRat(ref = refHe, SEref = refSEHe, obs = He,
                           SEobs = SEHe, type = "He"))
adjusted_na <- cbind(adjusted_na, Ar.na, Her.na)
adjusted_na[, c("Population","Ar","SEAr","Her","SEHer")]
@

To compare Cantabrian bears to the populations in North America, we also calculate
reference-population calibrated ratios for this population, and we have comparable
genetic diversity indices even if different locus panels and different sample sizes
were used. Result is presented in table \ref{table:pop}.

<<cantabrian1, echo = TRUE, eval = FALSE, tidy = FALSE>>=
cant.pops <- bear.diversity[bear.diversity$Study == 7, ]
loci_cant <- c("L02", "L03", "L04", "L05", "L06", "L08", "L09",
               "L10", "L11", "L12", "L13", "L18", "L19", "L20")
adjusted_cant <- runall(N = cant.pops$N,
                        genotypes = dinaric.genotypes,
                        loci = loci_cant,
                        nboots = 100)
names(adjusted_cant) <- paste("ref", names(adjusted_cant), sep = "")
adjusted_cant <- cbind(cant.pops, adjusted_cant)

Ar.cant <- with(adjusted_cant,
                calcDivRat(ref = refA, SEref = refSEA, obs = A,
                        SEobs = SEA, type = "A"))
Her.cant <- with(adjusted_cant,
                 calcDivRat(ref = refHe, SEref = refSEHe, obs = He,
                         SEobs = SEHe, type = "He"))
adjusted_cant <- cbind(adjusted_cant, Ar.cant, Her.cant)

adjusted_na_out <- adjusted_na[, c("Population", "Ar", "SEAr",
                                   "Her", "SEHer")]
names(adjusted_na_out) <- c("Population", "A", "SEA",
                            "He", "SEHe")
pops.comparison <- rbind(
          adjusted_na_out,
          adjusted_cant[, c("Population", "A", "SEA",
                            "He", "SEHe")])

pops.comparison
@

Look at the population comparison with comparable diversity indices:
<<cantabrian2, echo = FALSE, eval = TRUE, results = "asis">>=
print(xtable(pops.comparison, caption = "reference-population calibrated ratios for this population",
    label = "table:pop"), caption.placement = "top", table.placement = "!h")
@

\newpage
\section*{Results from the paper}

\subsection*{North America^{\cite{paetkau1998a, paetkau1998b}}}
\texttt{Nsamples\_usa} is a vector of the number of samples.

<<usa1, eval = FALSE, tidy = FALSE>>=
loci_usa <- c("L02", "L03", "L04", "L06",
              "L07", "L08", "L09", "L10")    
Nsamples_usa <- c(28, 50, 119, 148, 40, 41, 55, 45,
                  58, 30, 36, 57, 35, 34)  
adjusted_usa <- runall(N = Nsamples_usa,
                      genotypes = dinaric.genotypes,
                      loci = loci_usa, nboots = 1000)
@

<<usa2, echo = TRUE>>=
adjusted_usa
@

\subsection*{Scandinavia^{\cite{waits2000}}}

<<scan1, eval = FALSE, tidy = FALSE>>=
loci_skandinavija <- c("L02", "L03", "L04", "L05", "L06", "L07",
                       "L08", "L09", "L10", "L11", "L13", "L15",
                       "L17", "L18", "L19", "L20")
Nsamples_skand <- c(108, 29, 155, 88)
adjusted_skand <- runall(N = Nsamples_skand,
                         genotypes = dinaric.genotypes,
                         loci = loci_skandinavija,
                         nboots = 1000)
@
<<scan2, echo = TRUE>>=
adjusted_skand
@

\subsection*{Romania and Ital^{\cite{zachos2008}}}

<<roma1, eval = FALSE, tidy = FALSE>>=
loci_RO_I <- c("L02","L03","L04","L06","L08",
               "L10","L15","L18","L19")
Nsamples_ROI <- c(16, 17)
adjusted_ROI <- runall(N = Nsamples_ROI,
                       genotypes = dinaric.genotypes,
                       loci = loci_RO_I,
                       nboots = 1000)
@

<<roma2>>=
adjusted_ROI	
@

\subsection*{Cantabria^{\cite{perez2009}}}
<<cant1, eval = FALSE, tidy = FALSE>>=
loci_Cantabria <- c("L02", "L03", "L04", "L05", "L06", "L08", 
                    "L09", "L10", "L11", "L12", "L13", "L18",
                    "L19", "L20")
Nsamples_Cant <- c(8, 39)
adjusted_Cant <- runall(N = Nsamples_Cant,
                        genotypes = dinaric.genotypes,
                        loci = loci_Cantabria,
                        nboots = 1000)
@

<<cant2>>=
adjusted_Cant
@

\subsection*{Pakistan^{\cite{bellemain2006}}}
<<pak1, eval = FALSE, tidy = FALSE>>=
loci_Pakistan <- c("L02", "L03", "L04", "L05", "L06", "L09", 
                   "L10", "L13", "L15", "L17", "L18", "L19")
Nsamples_Pak <- 28
adjusted_pak <- runall(N = Nsamples_Pak,
                       genotypes = dinaric.genotypes,
                       loci = loci_Pakistan,
                       nboots = 1000)
@

<<pak2>>=
adjusted_pak
@

\subsection*{Greece^{\cite{karamanlidis2010}}}

<<grc1, eval = FALSE, tidy = FALSE>>=
loci_Greece <- c("L03", "L04", "L05", "L08", "L17", "L19")
Nsamples_Greece <- 49
adjusted_Greece <- runall(N = Nsamples_Greece,
                          genotypes = dinaric.genotypes,
                          loci = loci_Greece,
                          nboots = 1000)
@

<<grc2>>=
adjusted_Greece
@

\subsection*{Croatia^{\cite{kocijan2011}}}
<<cro1, eval = FALSE, tidy = FALSE>>=
loci_Croatia <- c("L02", "L03", "L04", "L05", "L06", "L08",
                  "L09", "L13", "L17", "L18", "L19")
Nsamples_Croatia <- 156
adjusted_Croatia <- runall(N = Nsamples_Croatia,
                           genotypes = dinaric.genotypes,
                           loci = loci_Croatia,
                           nboots = 1000)
@

<<cro2>>=
adjusted_Croatia
@

\subsection*{Slovakia and Romania^{\cite{straka}}}
<<slor1, eval = FALSE, tidy = FALSE>>=
loci_SkRo <- c("L02", "L03", "L04", "L05", "L06", "L07", 
               "L08", "L09", "L13", "L17", "L18", "L19")
Nsamples_SkRo <- c(71,96,16,109)
adjusted_SkRo <- runall(N = Nsamples_SkRo,
                        genotypes = dinaric.genotypes,
                        loci = loci_SkRo,
                        nboots = 1000)
@

<<slor2>>=
adjusted_SkRo
@

\subsection*{Gobi^{\cite{mccarthy2009}}}
<<gobi1, eval = FALSE, tidy = FALSE>>=
loci_gobi <- c("L02", "L03", "L04", "L06", "L09", "L10")
Nsamples_gobi <- 8
adjusted_gobi <- runall(N = Nsamples_gobi,
                        genotypes = dinaric.genotypes,
                        loci = loci_gobi,
                        nboots = 1000)
@

<<gobi2>>=
adjusted_gobi
@

\begin{thebibliography}{99}
\bibitem{skrbinsek2012} Skrbinšek T, Jelenčič M, Waits LP, Potočnik H, Kos I, Trontelj P(2012) Using a reference population yardstick to calibrate and compare genetic diversity reported in different studies: an example from the brown bear. Heredity, In press.
\bibitem{paetkau1998a} Paetkau DW, Shields GF, Strobeck C (1998) Gene flow between insular, coastal and interior populations of brown bears in Alaska. Molecular Ecology, 7, 1283-1292.
\bibitem{paetkau1998b} Paetkau DW, Waits LP, Clarkson PL, Craighead L, Vyse E, Ward R, Strobeck C (1998) Variation in Genetic Diversity across the Range of North American Brown Bears. Conservation Biology, 12, 418-429.
\bibitem{waits2000} Waits LP, Taberlet P, Swenson JE, Sandegren F, Franz R (2000) Nuclear DNA microsatellite analysis of genetic diversity and gene flow in the Scandinavian brown bear (\emph{Ursus arctos}). Molecular Ecology, 9, 421-431.
\bibitem{zachos2008} Zachos FE, Otto M, Unici R, Lorenzini R, Hartl GB (2008) Evidence of a phylogeographic break in the Romanian brown bear (\emph{Ursus arctos}) population from the Carpathians. Mammalian Biology - Zeitschrift fur Saugetierkunde, 73, 93-101.
\bibitem{perez2009} Pérez T, Vázquez F, Naves J, Fernández A, Corao A, Albornoz J, Domínguez A (2009) Non-invasive genetic study of the endangered Cantabrian brown bear (\emph{Ursus arctos}). Conservation Genetics, 10, 291-301.
\bibitem{bellemain2006} Bellemain E, Nawaz MA, Valentini A, Swenson JE, Taberlet P (2006) Genetic tracking of the brown bear in northern Pakistan and implications for conservation. Biological Conservation, 134, 537-547.
\bibitem{karamanlidis2010} Karamanlidis A, Drosopoulou E, de Gabriel Hernando M, Georgiadis L, Krambokoukis L, Pllaha S, Zedrosser A, Scouras Z (2010) Noninvasive genetic studies of brown bears using power poles. European Journal of Wildlife Research, 56, 693-702.
\bibitem{kocijan2011} Kocijan I, Galov A, Ćetković H, Kusak J, Gomerčić T, Huber Ä (2011) Genetic diversity of Dinaric brown bears (\emph{Ursus arctos}) in Croatia with implications for bear conservation in Europe. Mammalian Biology - Zeitschrift fur Saugetierkunde, 76, 615-621.
\bibitem{straka} Straka M, Paule L, Ionescu O, Štofík J, Adamec M (letnica) Microsatellite diversity and structure of Carpathian brown bears (\emph{Ursus arctos}): consequences of human caused fragmentation. Conservation Genetics, 1-12.
\bibitem{mccarthy2009} McCarthy TM, Waits LP, Mijiddorj B (2009). Status of the Gobi bear in Mongolia as determined by noninvasive genetic methods. Ursus 20(1): 30-38.
\end{thebibliography}

%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%
% tale del je tu zato, da izpiÅ¡e v katerem R-u se je delalo
% kamot vrÅ¾eÅ¡ ven, bo Å¡e vedno delalo
\newpage
<<>>=
sessionInfo()
@
%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%

\end{document}